pkg/crc/machine/machine.go

```diff
@@ -582,6 +582,8 @@ func Status(statusConfig ClusterStatusConfig) (ClusterStatusResult, error) {
 				openshiftVersion = crcBundleMetadata.GetOpenshiftVersion()
 			}
 			openshiftStatus = fmt.Sprintf("Running (v%s)", openshiftVersion)
+		} else {
+			openshiftStatus = "Starting"
```

Comment:
`func GetClusterOperatorStatus(oc OcConfig) (bool, error)` should return a more fine-grained status than just a boolean, if at least one operator is `Degraded`, I'm not sure `Starting` is correct.

Reply:
@cfergeau isn't it always at least one operator when the return value from `GetClusterOperatorStatus` is not false or you want which all are in degraded state?

Reply:
if we want to be finegrained we probably need to determine this ourselves.

Reply:
`false` can mean "Starting" if one or more operators are "progressing", but it could also mean "Degraded" if one or more operators are "degraded". So hardcoding that `false` means "starting" in `machine.go` does not seem very good, I prefer that we do it in `GetClusterOperatorStatus` where all the logic already is.

Reply:
@cfergeau you want to have `Degraded` as `stopped` condition? As of now there can be following conditions and I want to understand your prospective on those?
- Available => True  (everything working)
- Degraded => An operator reports Degraded when its current state does not match its desired state over a period of time resulting in a lower quality of service.  (What you prefer here? ). Does this mean openshift is Stopped or in Starting state (since we are not using any intermediate state). 
 Ref: https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md
- Progressing => Starting

Do we want an intermediate state which says `Degraded` as whole cluster, if that is the case then we need to put some refactor, otherwise for me a `false` return from `GetClusterOperatorStatus` means it is starting.

Reply:
I don't want to hardcode in `machine.go` that if `GetClusterOperatorStatus` returns `false`, then it means that the cluster is still being started. This is not always going to be true, as as you say, there's also 'degraded' to encode, and also potentially 'stopped'.

`GetClusterOperatorStatus` should return an enum instead of a boolean.
I'm fine if `GetClusterOperatorStatus` is changed for now to only report `starting` in addition to `started` while we figure out what to do with `degraded`.

Reply:
@cfergeau https://github.com/code-ready/crc/pull/1058/commits/eb6373f7d5562f325f1f6d29df49cd9616b56ca3 is what I come up with, now we don't skip any of the cluster state and make use of it.